src/requireAuth.ts

```diff
@@ -52,7 +50,10 @@ async function autoAuth(options: Options, authScopes: string[]): Promise<void |
     // Make sure any error here doesn't block the CLI, but log it.
     logger.debug(`Error getting account credentials.`);
   }
-
+  if (process.env.MONOSPACE_ENV && token && clientEmail) {
```

Comment:
Note: we should maybe replace this MONOSPACE check with a check whether clientEmail is a user email (as opposed to a service account email). IIUC, this would make ADCs set through GCloud work more consistently with some of our login flows. Likely do need to iron out some details around logout with it tho

Reply:
++ we'll need a huge rework on logout anyways i think...